### DISCUSSION TITLE
No of ongoing session

### DISCUSSION THREAD

**Discussion by k00lkd** [coturn/coturn#1091] [General]
How we can know how many session are currently live. Is there any way we can see data of no. of stun and turn sessions established everyday.

**Comment by eakraly** [comment]
If you use turnserver with prometheus then `turn_total_allocations` will give you current number of allocations
With that you can build a graph

I have much less experince with redis as a monitoring solution so not sure about that
Another approach you can take (somewhat hacky and may be unreliable) - extract this info from logs (count some specific lines)

**Reply by k00lkd** [reply]
Hi @eakraly ,I tried with Prometheus API but I couldn't find turn_total_allocations . Is there any plugin or anything extra add-on we need to install to get turn server related informations.

**Reply by eakraly** [reply]
Do you see any of the turn metrics in the prometheus response?
What's the output that you get?

**Reply by k00lkd** [reply]
I couldn't see any of the turn metrics. It is giving only system related metrics. 
I'm attaching output in a text file here.
[prometheus api output.txt](https://github.com/coturn/coturn/files/10294045/prometheus.api.output.txt)

**Reply by eakraly** [reply]
@k00lkd I think you are scraping the wrong interface - this output looks like the output of node_exporter that runs on the machine

You need to scrape the turnserver process directly (on prom port, default 9641)
See some details here https://github.com/coturn/coturn/blob/master/README.turnserver#L288

**Reply by k00lkd** [reply]
Thanks @eakraly  for response. I will update after trying this.

**Reply by k00lkd** [reply]
I tried with port 9641, prometheus is not connecting with this port to collect metrics

**Reply by k00lkd** [reply]
Hi @eakraly  Please check this https://github.com/coturn/coturn/issues/785  
Seems like we need to have some plugin installed. Can you please guide

**Reply by eakraly** [reply]
1. Make sure you run with `--prometheus`
2. If you get an error about unknown argument `--prometheus` then your turnserver is compiled without prometheus support
3. Pls share logs so we can assist you

**Reply by k00lkd** [reply]
@eakraly we installed prometheus in an server and now I'm trying to configure it to connect to turn server and get the metrics. Seems like I'm missing something. I want to know which package need to install to support Prometheus with turnserver.

[premetheus_response.txt](https://github.com/coturn/coturn/files/10318812/premetheus_response.txt)

**Reply by k00lkd** [reply]
hi @eakraly  Have you seen the attached response

**Reply by eakraly** [reply]
@k00lkd you are attaching log from prometheus itself

You should configure prometheus to scrape turnserver on 9641
In order to test if turnserver exposes its metrics you can `curl http://turnserver_ip:9641/metrics`
This should return you all the metrics in prometheus format

Setting up the whole prometheus server is beyond what I can help you with

**Comment by chole488** [comment]
We encountered the same problem, installed Prometheus and configured in order to collect turnserver metrics via the 9641 port. but not collecting the metrics, 
how to expose 9641 port, which exporter need to install to expose 9641 port.
Please direct us.

**Reply by eakraly** [reply]
You do not need to install any exporter
You need to properly configure prometheus itself - add new target (turnserver)
Prometheus does not auto-magically discover new scraping targets (remote or local)

If you can `curl http://turnserver_ip:9641/metrics` (or whatever port you expose it on) then turnserver part is fine

**Reply by adc-tgupta** [reply]
I am not seeing any metrics like turn_total_allocations . Is there any list of metrics which we should see with prometheus?
